docs/best-practices/json_type.md

```diff
@@ -0,0 +1,312 @@
+---
+slug: /best-practices/use-json-where-appropriate
+sidebar_position: 10
+sidebar_label: 'Using JSON'
+title: 'Use JSON where appropriate'
+description: 'Page describing when to use JSON'
+---
+
+ClickHouse now offers a native JSON column type designed for semi-structured and dynamic data. It’s important to clarify that **this is a column type, not a data format**—you can insert JSON into ClickHouse as a string or via supported formats like [JSONEachRow](/docs/interfaces/formats/JSONEachRow), but that does not imply using the JSON column type. Users should only use the JSON type when the structure of their data is dynamic, not when they simply happen to store JSON.
+
+## When to Use the JSON Type {#when-to-use-the-json-type}
+
+Use the JSON type when your data:
+
+* Has **unpredictable keys** that can change over time.
+* Contains **values with varying types** (e.g., a path might sometimes contain a string, sometimes a number).
+* Requires schema flexibility where strict typing isn’t viable.
+
+If your data structure is known and consistent, there is rarely a need for the JSON type, even if your data is in JSON format. Specifically, if your data has:
+
+* **A flat structure with known keys**: use standard column types e.g. String.
+* **Predictable nesting**: use Tuple, Array, or Nested types for these structures.
+* **Predictable structure with varying types**: consider Dynamic or Variant types instead.
+
+You can also mix approaches - for example, use static columns for predictable top-level fields and a single JSON column for a dynamic section of the payload.
+
+## Considerations and Tips for Using JSON {#considerations-and-tips-for-using-json}
+
+The JSON type enables efficient columnar storage by flattening paths into subcolumns. But with flexibility comes responsibility. To use it effectively:
+
+* **Specify path types** using [hints in the column definition](/sql-reference/data-types/newjson) to specify types for known sub columns, avoiding unnecessary type inference. 
```

Comment:
Type hints not only avoid unnecessary type inference but their main advantage (not mentioned in the text) is that they completely avoid any indirection (e.g. via discriminator columns) when the data is stored and processed as JSON paths with type hints are stored exactly like traditional types or exactly as if these paths would have been extracted into separate top-level fields. That also touches the recommendation at the beginning to avoid JSON if the data is known and consistent - with type hints then there is no need to adjust ingest and schema etc - convenient and still as performant as if we would have created top level fields for all JSON paths.